src/auto-update-manager.js

```diff
@@ -0,0 +1,84 @@
+'use babel'
+
+import {Emitter, CompositeDisposable} from 'event-kit'
+import {ipcRenderer} from 'electron'
+
+export default class AutoUpdateManager {
+  constructor ({applicationDelegate}) {
+    this.subscriptions = new CompositeDisposable()
+    this.emitter = new Emitter()
+
+    this.subscriptions.add(
+      applicationDelegate.onDidBeginCheckingForUpdate(() => {
+        this.emitter.emit('did-begin-checking-for-update')
+      }),
+      applicationDelegate.onDidBeginDownloadingUpdate(() => {
+        this.emitter.emit('did-begin-downloading-update')
+      }),
+      applicationDelegate.onDidCompleteDownloadingUpdate((details) => {
+        this.emitter.emit('did-complete-downloading-update', details)
+      }),
+      applicationDelegate.onUpdateNotAvailable(() => {
+        this.emitter.emit('update-not-available')
+      })
+    )
+  }
+
+  dispose () {
+    this.subscriptions.dispose()
+    this.emitter.dispose()
+  }
+
+  checkForUpdate () {
+    ipcRenderer.send('check-for-update')
+  }
+
+  restartAndInstallUpdate () {
+    ipcRenderer.send('install-update')
```

Comment:
Technically, these interactions should also be routed through the application delegate. Its purpose is to stand in for all interactions with the global application state which facilitates testing and improves isolation.